src/test/java/commaproject/be/commaserver/service/CommaServiceTest.java

```diff
@@ -0,0 +1,125 @@
+
+@ExtendWith(MockitoExtension.class)
+class CommaServiceTest {
+
+    @InjectMocks
+    private CommaServiceImpl commaService;
+
+    @Mock
+    private CommaRepository commaRepository;
+
+    @Test
+    @DisplayName("DB에 저장된 회고를 조회하면 모든 회고를 조회하고 테스트가 성공한다")
+    void findAll() {
+        // given
+        List<Comma> commas = setCommaData();
+        when(commaRepository.findAll()).thenReturn(commas);
+
+        // when
+        List<CommaDetailResponse> commaDetailResponsesExpected = commaService.readAll();
+
+        // then
+        assertThat(commaDetailResponsesExpected.size()).isEqualTo(3);
+    }
+
+
+    @Test
+    @DisplayName("유효한 commaId로 회고를 조회하면 테스트가 성공한다")
+    void valid_comma_id_find_comma() {
+        // given
+        Long commaId = 1L;
+        Long userId = 1L;
+        Optional<Comma> comma = Optional.of(Comma.from("title1", "content1", "username1", userId));
+        when(commaRepository.findById(commaId)).thenReturn(comma);
+
+        // when
+        CommaDetailResponse commaDetailResponse = commaService.readOne(commaId);
+
+        // then
+        // todo id를 조회하고자 했는데 comma를 jpa로 저장하지 않아서 id 생성이 안되는걸까?
+        assertThat(commaDetailResponse).isNotNull();
+        assertThat(commaDetailResponse.getTitle()).isEqualTo("title1");
+    }
+
+    @Test
+    @DisplayName("유효한 commaId로 회고를 조회하여 수정한 데이터를 확인할 수 있으면 테스트가 성공한다")
+    void update_comma() {
+        // given
+        Long commaId = 1L;
+        Long userId = 1L;
+        Optional<Comma> comma = Optional.of(Comma.from("title1", "content1", "username1", userId));
+        when(commaRepository.findById(commaId)).thenReturn(comma);
+
+        // when
+        CommaRequest commaRequest = new CommaRequest("title1", "update content1", "username", userId
+        );
+        CommaDetailResponse updateCommaDetailResponse = commaService.update(commaId, commaRequest);
+
+        // then
+        assertThat(updateCommaDetailResponse.getContent()).isEqualTo("update content1");
+    }
+
+    @Test
+    @DisplayName("회고 데이터가 저장되면 테스트가 성공한다")
+    void save_comma() {
+        // given
+        Long userId = 1L;
+        Comma comma = Comma.from("title1", "content1", "username", userId);
+        when(commaRepository.save(any(Comma.class))).thenReturn(comma);
+
+        // when
+        CommaResponse saveCommaResponse = commaService.create(
+            new CommaRequest("title1", "content1", "username", userId));
+
+        // then
+        assertThat(saveCommaResponse).isNotNull();
+    }
+
+    @Test
+    @DisplayName("유효한 commaId로 회고를 조회하여 soft delete를 하면 테스트가 성공한다")
+    void remove_comma() {
+        // given
+        Long userId = 1L;
+        Comma comma = Comma.from("title1", "content1", "username", userId);
+        when(commaRepository.save(any(Comma.class))).thenReturn(comma);
+
+
+        // when
+        Long commaId = 1L;
+        commaService.remove(commaId);
+
+        // then
+        verify(commaRepository, times(1)).save(any(Comma.class));
+    }
```

Comment:
mock 했기 때문에 save()메서드 호출시에만 저장됩니다.
따라서 자동으로 데이터가 저장되지 않습니다.

구현하신 remove()메서드는 아래와 같은데
```java
    public CommaResponse remove(Long commaId) {
        Comma findComma = commaRepository.findById(commaId)
            .orElseThrow(NoSuchElementException::new); // findById() 메서드에 대한 mock이 없기 때문에 null이 발생하고, 자연스럽게 NoSuch~ 테스트 실패가 아닌 익셉션이 발생했습니다.
        commaRepository.delete(findComma);
        return new CommaResponse(findComma.getId());
    }
```

여기서 검증하고자 하는 것이 delete 메서드가 1번 실행되었는가? 검증하는 것이라면 테스트 코드는 아래와 같은 예로 작성되어야 합니다.
```suggestion
    @Test
    @DisplayName("유효한 commaId로 회고를 조회하여 soft delete를 하면 테스트가 성공한다")
    void remove_comma() {
        // given
        Long userId = 1L;
        Comma comma = Comma.from("title1", "content1", "username", userId);
        when(commaRepository.findById(any())).thenReturn(Optional.of(comma));


        // when
        Long commaId = 1L;
        commaService.remove(commaId);

        // then
        verify(commaRepository, times(1)).delete(any());
    }

```

Reply:
`mock`하면 데이터를 저장하고 저장한 데이터를 호출할 수 없군요 ㅎㅎ;; 감사합니다!!

Reply:
mock은 말그대로 mock 이니까요.
JPA나 레포 구현체는 작동하지 않죠

Reply:
JPA가 해주는 것들로 테스트하기 위해선 통합테스트가 필요하겠군요 🤔

Reply:
JPA만을 테스트하기 원한다면 테스트 컨텍스트 범위를 줄이기 위해 `@DataJpaTest` 사용할 수 있어요.

Reply:
통합테스트를 할 때 `@DataJpaTest`로 테스트 컨텍스트 범위 조정!!! 감사합니다 ㅎㅎ